manifests/controller.pp

```diff
@@ -111,6 +116,20 @@
     },
   }
 
+  if ! $controller_worker {
+    # Make Puppet stop the mesos-slave service
+    service { 'mesos-slave':
+      ensure  => stopped,
+      require => Package['mesos'],
+    }
+
+    # Stop mesos-slave service from starting at startup
+    file { '/etc/init/mesos-slave.override':
+        content => 'manual',
+        notify  => Service['mesos-slave']
+    }
```

Comment:
Hrm.. it's kind of a pain.

We haven't tested switching roles of already-provisioned machines but I suppose it would be good if it did work...